test/visibility_test/visibility_test.go

```diff
@@ -0,0 +1,527 @@
+					ctx.SMWithOAuth.POST("/v1/visibilities").
+						WithJSON(common.Object{
+							"service_plan_id": existingPlanIDs[0],
+						}).
+						Expect().Status(http.StatusCreated)
+
+					ctx.SMWithOAuth.POST("/v1/visibilities").
+						WithJSON(common.Object{
+							"service_plan_id": existingPlanIDs[0],
+							"platform_id":     existingPlatformID,
+						}).
+						Expect().Status(http.StatusBadRequest).JSON().Object().Keys().Contains("error", "description")
+
+				})
+			})
+
+			Context("when a record with the same or null platform id does not exist", func() {
+				It("returns 201", func() {
+					ctx.SMWithOAuth.POST("/v1/visibilities").
+						WithJSON(postVisibilityRequest).
+						Expect().Status(http.StatusCreated).JSON().Object().ContainsMap(postVisibilityRequest).Keys().Contains("id")
+				})
+			})
+		})
+	})
+
+	Describe("PATCH", func() {
+		var existingVisibilityID string
+		var anotherExistingPlatformID string
+
+		BeforeEach(func() {
+			existingVisibilityID = ctx.SMWithOAuth.POST("/v1/visibilities").
+				WithJSON(common.Object{
+					"platform_id":     existingPlatformID,
+					"service_plan_id": existingPlanIDs[0],
+				}).
+				Expect().Status(http.StatusCreated).JSON().Object().Value("id").String().Raw()
+
+			anotherPlatform := ctx.RegisterPlatform()
+			anotherExistingPlatformID = anotherPlatform.ID
+		})
+
+		Context("when updating properties with valid values", func() {
+			It("returns 200", func() {
+				ctx.SMWithOAuth.PATCH("/v1/visibilities/" + existingVisibilityID).
+					WithJSON(common.Object{
+						"platform_id":     anotherExistingPlatformID,
+						"service_plan_id": existingPlanIDs[1],
+					}).
+					Expect().
+					Status(http.StatusOK).JSON().Object().ContainsMap(common.Object{
+					"id":              existingVisibilityID,
+					"platform_id":     anotherExistingPlatformID,
+					"service_plan_id": existingPlanIDs[1],
+				})
+
+				ctx.SMWithOAuth.GET("/v1/visibilities/" + existingVisibilityID).
+					Expect().
+					Status(http.StatusOK).JSON().Object().ContainsMap(common.Object{
+					"id":              existingVisibilityID,
+					"platform_id":     anotherExistingPlatformID,
+					"service_plan_id": existingPlanIDs[1],
+				})
+			})
+		})
+
+		Context("when created_at is in the body", func() {
+			It("should not update created_at", func() {
+				createdAt := "2015-01-01T00:00:00Z"
+
+				ctx.SMWithOAuth.PATCH("/v1/visibilities/"+existingVisibilityID).
+					WithJSON(common.Object{
+						"created_at": createdAt,
+					}).
+					Expect().
+					Status(http.StatusOK).JSON().Object().
+					ContainsKey("created_at").
+					ValueNotEqual("created_at", createdAt)
+
+				ctx.SMWithOAuth.GET("/v1/visibilities/"+existingVisibilityID).
+					Expect().
+					Status(http.StatusOK).JSON().Object().
+					ContainsKey("created_at").
+					ValueNotEqual("created_at", createdAt)
+			})
+		})
+
+		Context("when updated_at is in the body", func() {
+			It("should not update created_at", func() {
```

Comment:
"should not update updated_at"

Also, maybe add a check that updated_at is changed, but is not the one provided in the body

Reply:
the check is there - line 436 - ValueNotEqual("updated_at", updatedAt) - or do you have something else in mind ?

Reply:
I meant GET /visibilities/<visibility_id> and get the updated_at field before the update. Update the visibility. execute GET/visibilities/<visibility_id> and validate that the updated_at is different that the one before the update.

Reply:
problem is no time passes between the requests, so we need to sleep which im not sure is very good option

```
updatedAt := "2015-01-01T00:00:00Z"

				currentUpdateAtValue := ctx.SMWithOAuth.GET("/v1/visibilities/" + existingVisibilityID).
					Expect().
					Status(http.StatusOK).JSON().Object().Value("updated_at").String().Raw()

				time.Sleep(1 * time.Second) // without this, the test would fail
				ctx.SMWithOAuth.PATCH("/v1/visibilities/"+existingVisibilityID).
					WithJSON(common.Object{
						"updated_at": updatedAt,
					}).
					Expect().
					Status(http.StatusOK).JSON().Object().
					ContainsKey("updated_at").
					ValueNotEqual("updated_at", updatedAt).
					ValueNotEqual("updated_at", currentUpdateAtValue)

				ctx.SMWithOAuth.GET("/v1/visibilities/"+existingVisibilityID).
					Expect().
					Status(http.StatusOK).JSON().Object().
					ContainsKey("updated_at").
					ValueNotEqual("updated_at", updatedAt).
					ValueNotEqual("updated_at", currentUpdateAtValue)
			})
```